src/SSLSocket.c

```diff
@@ -552,23 +552,31 @@ int SSLSocket_setSocketForSSL(networkHandles* net, MQTTClient_SSLOptions* opts)
 	if (net->ctx != NULL || (rc = SSLSocket_createContext(net, opts)) == 1)
 	{
 		int i;
-		SSL_CTX_set_info_callback(net->ctx, SSL_CTX_info_callback);
-		SSL_CTX_set_msg_callback(net->ctx, SSL_CTX_msg_callback);
-   		if (opts->enableServerCertAuth) 
-			SSL_CTX_set_verify(net->ctx, SSL_VERIFY_PEER, NULL);
+		if (net->ctx != NULL)
```

Comment:
I've run into this problem too. I ran into it when accidentally using an incorrect private key password. I was convinced that the problem was in SSLSocket_createContext() itself, and that if SSL_CTX_use_PrivateKey_file() failed the return code ``rc`` needed to be set, something like this:

```c
        rc1 = SSL_CTX_use_PrivateKey_file(net->ctx, opts->privateKey, SSL_FILETYPE_PEM);
        if (opts->privateKey == opts->keyStore)
            opts->privateKey = NULL;
        if (rc1 != 1)
        {
            rc = rc1; // rc will be != 1, and callers will not use ctx
            SSLSocket_error("SSL_CTX_use_PrivateKey_file", NULL, net->socket, rc);
            goto free_ctx;
        }  
```

It seems odd to me that a new ``rc1`` was created at all, but I can only assume the author had a good reason for it.

That's not to say that the caller shouldn't double-check to make sure SSLSocket_createContext() hasn't invalidated ``ctx``, but maybe it's worth making the function behave according to expectations as well.